File: tools/figcheck/fixtures/matplotlib/good.py
"""Minimal matplotlib script for figcheck's matplotlib backend self-test.

Writes good.png into $FIGCHECK_OUTDIR (or the working directory).
"""

import os

import matplotlib

matplotlib.use("Agg")
import matplotlib.pyplot as plt

outdir = os.environ.get("FIGCHECK_OUTDIR", ".")
os.makedirs(outdir, exist_ok=True)

fig, ax = plt.subplots(figsize=(2, 2))
ax.plot([0, 1, 2], [0, 1, 4])
ax.set_xlabel("x")
ax.set_ylabel("y")
fig.savefig(os.path.join(outdir, "good.png"))

File: tools/figcheck/fixtures/makie/good.jl
# Minimal CairoMakie script for figcheck's makie backend self-test.
# Writes good.png into ENV["FIGCHECK_OUTDIR"] (or the working directory).
using CairoMakie

outdir = get(ENV, "FIGCHECK_OUTDIR", ".")
mkpath(outdir)

fig = Figure(size = (200, 200))
ax = Axis(fig[1, 1], xlabel = "x", ylabel = "y")
lines!(ax, [0, 1, 2], [0, 1, 4])
save(joinpath(outdir, "good.png"), fig)
